--- src/tty_interface.c	2021-07-30 02:19:58.574493751 +0300
+++ src/tty_interface.c.patched	2021-07-30 02:25:09.617945912 +0300
@@ -329,10 +329,10 @@
 					   {KEY_CTRL('T'), action_select},	 /* C-T */
 					   {KEY_CTRL('P'), action_prev},	 /* C-P */
 					   {KEY_CTRL('N'), action_next},	 /* C-N */
-					   {KEY_CTRL('K'), action_prev},	 /* C-K */
+					   {KEY_CTRL('E'), action_prev},	 /* C-K */
 					   {KEY_CTRL('J'), action_next},	 /* C-J */
 					   {KEY_CTRL('A'), action_beginning},    /* C-A */
-					   {KEY_CTRL('E'), action_end},		 /* C-E */
+					   {KEY_CTRL('K'), action_end},		 /* C-E */
 
 					   {"\x1bOD", action_left}, /* LEFT */
 					   {"\x1b[D", action_left}, /* LEFT */
